src/SessionSetup/RetrieveInstancesWidget.cpp

```diff
@@ -199,4 +219,44 @@ void RetrieveInstancesWidget::OnReloadButtonClicked() {
             });
 }
 
+std::optional<orbit_ggp::Project> RetrieveInstancesWidget::GetSelectedProject() {
+  std::optional<Project> project;
+  if (ui_->projectComboBox->currentData().canConvert<Project>()) {
+    project = ui_->projectComboBox->currentData().value<Project>();
+  }
+  return project;
+}
+
+void RetrieveInstancesWidget::OnProjectComboBoxCurrentIndexChanged() {
+  std::optional<Project> selected_project = GetSelectedProject();
+
+  // nothing changed, no loading required
+  if (selected_project == GetQSettingsProject()) return;
```

Comment:
Not sure if that is a reasonable assumption. What happens if a second Orbit instance changed the stored default project?

BTW, is that check really needed? When does QComboBox fires that signal and the current index didn't change?

Reply:
I removed the check after double checking that selecting the same entry does signal currentIndexChanged. This was mainly to catch this case.